modules/aws/sonar-base-instance/sg.tf

```diff
@@ -7,12 +7,14 @@ data "aws_vpc" "selected" {
 }
 
 locals {
-  cidr_blocks       = concat(var.sg_ingress_cidr, var.create_and_attach_public_elastic_ip ? try(["${aws_eip.dsf_instance_eip[0].public_ip}/32"], []) : [])
-  ingress_ports     = [22, 8080, 8443, 3030, 27117]
-  ingress_ports_map = { for port in local.ingress_ports : port => port }
+  should_create_security_group = var.security_group_id == "" ? true : false
+  cidr_blocks                  = var.sg_ingress_cidr
+  ingress_ports                = [22, 8080, 8443, 3030, 27117]
+  ingress_ports_map            = { for port in local.ingress_ports : port => port }
 }
 
 resource "aws_security_group" "dsf_base_sg" {
+  count       = local.should_create_security_group == true ? 1 : 0
```

Comment:
Please put this condition under locals{} and that local in all the `count`s below.